### ISSUE TITLE
[CI] ElasticsearchHostsSnifferTests#testSniffNodes test failure

### ISSUE THREAD

**Issue by martijnvg** [elastic/elasticsearch#26684]
Unfortunately this failure didn't print a reproduce line. Also I'm not able to reproduce it locally by re-running this test many times.

```
Suite: org.elasticsearch.client.sniff.ElasticsearchHostsSnifferTests
FAILURE 0.12s J0 | ElasticsearchHostsSnifferTests.testSniffNodes <<< FAILURES!
   > Throwable #1: java.lang.AssertionError: 
   > Expected: a string containing "GET http://127.0.0.1:46655/_nodes/http?timeout=1532ms"
   >      but: was "method [GET], host [http://127.0.0.1:46655], URI [/_nodes/http?timeout=1532ms], status line [HTTP/1.1 504 Gateway Timeout]
   > "
   > 	at __randomizedtesting.SeedInfo.seed([96843A38B3BE28EE:F7B56845D70E22E1]:0)
   > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   > 	at org.elasticsearch.client.sniff.ElasticsearchHostsSnifferTests.testSniffNodes(ElasticsearchHostsSnifferTests.java:131)
   > 	at java.lang.Thread.run(Thread.java:748)
```

Build url: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.0+multijob-unix-compatibility/os=centos/172/consoleText

**Comment by martijnvg** [comment]
Another instance: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.0+java9-periodic/637/console

**Comment by jasontedor** [comment]
Closed by 0814ea3200d6af6c142ad7d2fa16bce5acfa9665
